packages/objectloader2/src/operations/downloader.ts

```diff
@@ -0,0 +1,155 @@
+import BatchingQueue from '../helpers/batchingQueue.js'
+import Queue from '../helpers/queue.js'
+import { ObjectLoaderRuntimeError } from '../types/errors.js'
+import { isBase, Item } from '../types/types.js'
+import { ICache, IDownloader } from './interfaces.js'
+import { BaseDownloadOptions } from './options.js'
+
+export default class Downloader implements IDownloader {
+  private _serverUrl: string
+  private _streamId: string
+  private _objectId: string
+  private _token?: string
+  private _requestUrlRootObj: string
+  private _requestUrlChildren: string
+  private _headers: HeadersInit
+  private _options: BaseDownloadOptions
+
+  private _database: ICache
+  private _downloadQueue: BatchingQueue<string>
+  private _results: Queue<Item>
+
+  constructor(
+    database: ICache,
+    results: Queue<Item>,
+    serverUrl: string,
+    streamId: string,
+    objectId: string,
+    token?: string,
+    options?: Partial<BaseDownloadOptions>
+  ) {
+    this._database = database
+    this._results = results
+
+    this._serverUrl = serverUrl
+    this._streamId = streamId
+    this._objectId = objectId
+    this._token = token
+    this._options = {
+      ...{
+        fetch: (...args) => window.fetch(...args),
```

Comment:
1. you shouldn't reference `window` anywhere, this code has to be Node compatible
2. y not just `fetch: globalThis.fetch`

Reply:
I was going around in circles to make this work....I think my example just used window...can use the other I guess...will need to understand the difference though